capstone/src/main/java/com/google/sps/Objects/response/ChallengeResponse.java

```diff
@@ -0,0 +1,26 @@
+package com.google.sps.Objects.response;
+
+import com.google.sps.Objects.Challenge;
+
+/**
+ * This class is used in the ChallengeServlet to wrap together the current challenge and whether the
+ * current user has completed the challenge.
+ *
+ * @author lucyqu
+ */
+public class ChallengeResponse {
+
+  private Challenge challenge;
+  private boolean userHasCompleted;
+
+  /**
+   * Constructor to set the instance variables.
+   *
+   * @param challenge Challenge object
+   * @param isCompleted boolean indicating whether user has completed challenge
+   */
+  public ChallengeResponse(Challenge challenge, boolean userHasCompleted) {
+    this.challenge = challenge;
+    this.userHasCompleted = userHasCompleted;
```

Comment:
Sorry for being nitpicky, but boolean variables should start with is/has. I don't think you can see this g3doc so here are the guidelines:
By convention, boolean variables are usually named "foo" instead of "isFoo" so that it's a noun phrase rather than verb phrase just like most other variables. On the other hand, methods that return boolean are usually named "isFoo" so that it's a verb phrase just like most other methods.

I know you follow the noun phrase but we also want them to start the same

```suggestion
  private boolean hasUserCompleted;

  /**
   * Constructor to set the instance variables.
   *
   * @param challenge Challenge object
   * @param hasUserCompleted boolean indicating whether user has completed challenge
   */
  public ChallengeResponse(Challenge challenge, boolean hasUserCompleted) {
    this.challenge = challenge;
    this.hasUserCompleted = hasUserCompleted;
```